From: Stig Sandbeck Mathisen <ssm@debian.org>
Date: Sat, 10 May 2014 12:25:41 +0200
Subject: http_loadtime plugin: Fix several bugs
MIME-Version: 1.0
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

 - Missing «fi»

 - Unset variable for «wget» in «autoconf»

 - garbage to stdout from «cd -»
---
 plugins/node.d/http_loadtime.in | 21 ++++++++++-----------
 1 file changed, 10 insertions(+), 11 deletions(-)

diff --git a/plugins/node.d/http_loadtime.in b/plugins/node.d/http_loadtime.in
index 4364a02..a9d7854 100644
--- a/plugins/node.d/http_loadtime.in
+++ b/plugins/node.d/http_loadtime.in
@@ -42,14 +42,15 @@ if [ "$1" = "autoconf" ]; then
 	echo "no (programs time, wget and tr required)"
 	    exit 0
     fi
-    if ! $wget_bin -q -O /dev/null $target; then
-
-    # check if url responds
-    #
-    wget --spider $target $wget_opt
-    if [ "$?" != "0" ]; then
-        echo "no (Cannot run wget against \"$target\")"
-        exit 0
+    if ! wget -q -O /dev/null $target; then
+
+        # check if url responds
+        #
+        wget --spider $target $wget_opt
+        if [ "$?" != "0" ]; then
+            echo "no (Cannot run wget against \"$target\")"
+            exit 0
+        fi
     fi
     echo yes
     exit 0
@@ -70,8 +71,6 @@ TEMPO_DIR=$(mktemp -dt munin_http_loadtime.XXXXXX) || exit 1
 
 trap "rm -rf $TEMPO_DIR" EXIT
 
-cd $TEMPO_DIR || exit 1
-loadtime=`$time_bin --quiet -f "%e" wget $wget_opt $target 2>&1`
-cd -
+loadtime=$(cd $TEMPO_DIR && $time_bin --quiet -f "%e" wget $wget_opt $target 2>&1)
 
 echo "loadtime.value $loadtime"
